torch/onnx/symbolic_opset11.py

```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
+
+    # Cases when repeats is a single value tensor and dim has unknown input size
+    if (repeats_dim == 0 or (repeats_dim == 1 and repeats_sizes[0] == 1)) and input_sizes[dim] == 0:
+        if not sym_help._is_tensor(repeats):
+            repeats = g.op("Constant", value_t=torch.LongTensor(repeats))
+        reps = sym_help._size_helper(g, input, dim)
+        reps = unsqueeze(g, reps, 0)
+        repeats = g.op("Expand", repeats, reps)
+    elif repeats_dim == 1 and input_sizes[dim] == 0:
+        reps = sym_help._size_helper(g, input, dim)
+        reps = unsqueeze(g, reps, 0)
+    else:
+        return torch.onnx.symbolic_opset9.repeat_interleave(g, self, repeats, final_dim)
+
+    reps_like = g.op("ConstantOfShape", g.op("Shape", repeats),
+                     value_t=torch.tensor([1], dtype=torch.long))
+    r_splits = split(g, repeats, reps_like, 0)
+    i_splits = split(g, input, reps_like, dim)
+
+    input_sizes[dim], input_sizes_temp[dim] = -1, 1
+
+    # Create a loop to iterate over each value along the dimension
+    # and perform individual interleaving using the repeats tensor
+    # Loop is of the following pattern
+    # input (trip_count, cond)
+    #   int trip_count = ...;
+    #   bool cond = ...;
+    #   for (int i=0; i < trip_count && cond; ++i) {
+    #     cond = ...;
+    #   }
+
+    # Loop conditions
+    loop_condition = g.op("Constant", value_t=torch.tensor(1))
+    loop_condition = g.op("Cast", loop_condition, to_i=9)
+    loop_len = reps
+    loop = g.op("Loop", loop_len, loop_condition)
+
+    # Loop inputs
+    loop_block = _add_block(loop.node())
+    block_input_iter = _add_input_to_block(loop_block)
+    cond = _add_input_to_block(loop_block)
+
+    r_split = loop_block.op("SequenceAt", r_splits, block_input_iter)
+    i_split = loop_block.op("SequenceAt", i_splits, block_input_iter)
+
+    i_split = unsqueeze(loop_block, i_split, dim + 1)
+    r_concat = [loop_block.op("Constant", value_t=torch.LongTensor(input_sizes_temp[:dim + 1])),
+                r_split,
+                loop_block.op("Constant", value_t=torch.LongTensor(input_sizes_temp[dim + 1:]))]
+    r_concat = loop_block.op("Concat", *r_concat, axis_i=0)
+    i_split = expand(loop_block, i_split, r_concat, None)
+    i_split = reshape(loop_block, i_split, g.op("Constant", value_t=torch.LongTensor(input_sizes)))
+
+    # Loop outputs
+    cond_out = loop_block.op("Cast", loop_condition, to_i=9)
+    _add_output_to_block(loop_block, cond_out)
+    _add_output_to_block(loop_block, i_split)
+    loop_out = loop.node().output()
+
+    # In this loop, the outputs are scan outputs and are concatenated along
+    # the zero'th dimension (by default). In order to avoid this and concatenate
+    # along the dimension provided, some post-processing is required
+    loop_out = g.op("Transpose", loop_out, perm_i=perm_i)
+    return reshape(g, loop_out, g.op("Constant", value_t=torch.LongTensor(input_sizes)))
```

Comment:
nit: improve readability by renaming input_sizes and input_sizes_temp. e.g. for this line the input_sizes actually resembles output_sizes